src/rascal/math/kvec_generator.hh

```diff
@@ -120,14 +120,14 @@ namespace rascal {
        * row(i) = i-th vector. If the origin k=(0,0,0) is included, it is
        * always stored in row(0).
        */
-      Matrix_t get_kvectors() const {
+      Eigen::Ref<const Matrix_t> get_kvectors() {
         return Eigen::Ref<const Matrix_t>(this->kvectors);
       }
 
       /** Returns const reference to the Eigen vector containing the norm
        * of all k-vectors within the cutoff without double counting.
        */
-      Vector_t get_kvector_norms() const {
+      Eigen::Ref<const Vector_t> get_kvector_norms() {
```

Comment:
Ideally, the function should be marked as const if it returns a const reference.

```suggestion
      Eigen::Ref<const Vector_t> get_kvector_norms() const {
```